framework/src/loops/ComputeJacobianThread.C

```diff
@@ -83,14 +83,16 @@ ComputeJacobianThread::computeJacobian()
         kernel->computeJacobian();
   }
 
+  // TODO: I needed to remove some filtering criteria here to
+  // find the time kernel [in my example].
+  // The one that actually did the trick was removing the
+  // _tag condition. Not sure why. Needs investigation.
   std::vector<FVElementalKernel *> kernels;
   _fe_problem.theWarehouse()
              .query()
              .template condition<AttribSystem>("FVElementalKernel")
              .template condition<AttribSubdomains>(_subdomain)
              .template condition<AttribThread>(_tid)
-             .template condition<AttribVectorTags>(_tags)
-             .template condition<AttribIsADJac>(true)
```

Comment:
This needs to be fixed now rather than later.  We definitely need the ADJac attribute there or you will end up calling the RESIDUAL version of objects needlessly.  What time kernel were you using?

Reply:
I used `FVTimeKernel`.

Reply:
What input file can I run to see the problem?

Reply:
test/tests/kernels/fv_constant_scalar_advection/2D_constant_scalar_advection.i

Reply:
I think I have the fix.  I had just copied the query from the residual compute loop - but here we need to select on the matrix tags attribute rather than vector tags.

Reply:
I pushed up the fix to your branch.  Now we just need to iron out the class names and param details.